packages/block-library/src/latest-posts/edit.native.js

```diff
@@ -0,0 +1,229 @@
+import { __ } from '@wordpress/i18n';
+import { postList as icon } from '@wordpress/icons';
+import { InspectorControls } from '@wordpress/block-editor';
+import { fetchRequest } from 'react-native-gutenberg-bridge';
+import {
+	Icon,
+	PanelBody,
+	ToggleControl,
+	RangeControl,
+	QueryControls,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+import { MIN_EXCERPT_LENGTH, MAX_EXCERPT_LENGTH } from './constants';
+
+class LatestPostsEdit extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			categoriesList: [],
+		};
+	}
+
+	componentDidMount() {
+		this.isStillMounted = true;
+		this.fetchRequest = fetchRequest( '/wp/v2/categories' )
+			.then( ( categoriesList ) => {
+				if ( this.isStillMounted ) {
+					if ( typeof categoriesList === 'string' ) {
+						this.setState( {
+							categoriesList: JSON.parse( categoriesList ),
+						} );
+					} else {
+						this.setState( { categoriesList } );
+					}
+				}
+			} )
+			.catch( () => {
+				if ( this.isStillMounted ) {
+					this.setState( { categoriesList: [] } );
+				}
+			} );
+	}
+
+	componentWillUnmount() {
+		this.isStillMounted = false;
+	}
+
+	render() {
+		const {
+			attributes,
+			setAttributes,
+			getStylesFromColorScheme,
+			name,
+			openGeneralSidebar,
+			isSelected,
+		} = this.props;
+
+		const {
+			displayPostContent,
+			displayPostContentRadio,
+			excerptLength,
+			displayPostDate,
+			order,
+			orderBy,
+			postsToShow,
+			categories,
+		} = attributes;
+
+		const { categoriesList } = this.state;
+		const blockType = coreBlocks[ name ];
+		const displayExcerptPostContent = displayPostContentRadio === 'excerpt';
+
+		const blockStyle = getStylesFromColorScheme(
+			styles.latestPostBlock,
+			styles.latestPostBlockDark
+		);
+
+		const iconStyle = getStylesFromColorScheme(
+			styles.latestPostBlockIcon,
+			styles.latestPostBlockIconDark
+		);
+
+		const titleStyle = getStylesFromColorScheme(
+			styles.latestPostBlockMessage,
+			styles.latestPostBlockMessageDark
+		);
+
+		const onSetDisplayPostContent = ( value ) => {
+			setAttributes( { displayPostContent: value } );
+		};
+
+		const onSetDisplayPostContentRadio = ( value ) => {
+			setAttributes( {
+				displayPostContentRadio: value ? 'excerpt' : 'full_post',
+			} );
+		};
+
+		const onSetExcerptLength = ( value ) => {
+			setAttributes( { excerptLength: value } );
+		};
+
+		const onSetDisplayPostDate = ( value ) => {
+			setAttributes( { displayPostDate: value } );
+		};
+
+		const onSetOrder = ( value ) => {
+			setAttributes( { order: value } );
+		};
+
+		const onSetOrderBy = ( value ) => {
+			setAttributes( { orderBy: value } );
+		};
+
+		const onSetPostsToShow = ( value ) => {
+			setAttributes( { postsToShow: value } );
+		};
+
+		const onSetCategories = ( value ) => {
+			setAttributes( {
+				categories: '' !== value ? value.toString() : undefined,
+			} );
+		};
```

Comment:
I think by having all of these within the render function makes it too long.. What do you think if we move these out the `render` and just have them on its own, like:

```
	componentWillUnmount() {
		this.isStillMounted = false;
	}

	onSetExcerptLength = ( value ) => {
		const { setAttributes } = this.props;
		setAttributes( { excerptLength: value } );
	};

	onSetDisplayPostDate = ( value ) => {
		const { setAttributes } = this.props;
		setAttributes( { displayPostDate: value } );
	};

	onSetOrder = ( value ) => {
		const { setAttributes } = this.props;
		setAttributes( { order: value } );
	};

	render() {
```

And then just call them like

```
	<RangeControl
		label={ __( 'Excerpt length (words)' ) }
		value={ excerptLength }
		onChange={ this.onSetExcerptLength }
		min={ MIN_EXCERPT_LENGTH }
		max={ MAX_EXCERPT_LENGTH }
		separatorType="none"
	/>
```

Reply:
That works for me 👍

Reply:
Looks good :+1: . Not suggesting any changes here, just wanted to make a minor note that IIRC, web does not use `plugin-proposal-class-properties`, so if we merge `edit.js` and `edit.native.js` in some future iteration, we may need to bind these functions in the constructor.

Reply:
I'll have to adjust from this change a bit. I'm getting the following error from the `npm build` script:

`Support for the experimental syntax 'classProperties' isn't currently enabled`

Reply:
Yeah @mkevins is right sorry I completely forgot, you'd have to bind them =(

Reply:
No worries! I should have those adjusted here shortly testing them now.

Reply:
These changes are good to go now. I didn't recreate an app build for them though so you'll need to use metro to retest.

Reply:
I re-tested on Android (Pixel 3a) via WordPress-Android:develop and everything is still working as described. :tada: